src/modules/endpoint/DefaultDataContainer.js

```diff
@@ -8,20 +8,48 @@ import type { DataContainer } from './DataContainer'
 import type Moment from 'moment'
 import { difference, flatMap, isEmpty, map, omitBy } from 'lodash'
 import RNFetchBlob from 'rn-fetch-blob'
+import Cache from './Cache'
+
+type CacheType = {
+  events: Cache<Array<EventModel>>,
+  categories: Cache<CategoriesMapModel>,
+  languages: Cache<Array<LanguageModel>>,
+  resourceCache: Cache<CityResourceCacheStateType>,
+  lastUpdate: Cache<Moment>
+}
+
+type CacheKeyType = $Keys<CacheType>
 
 class DefaultDataContainer implements DataContainer {
   _databaseConnector: DatabaseConnector
-  _context: DatabaseContext | null = null
+  caches: CacheType
 
-  _lastUpdate: Moment | null
   _cities: Array<CityModel> | null
-  _categoriesMap: CategoriesMapModel | null
-  _languages: Array<LanguageModel> | null
-  _resourceCache: CityResourceCacheStateType | null
-  _events: Array<EventModel> | null
 
   constructor () {
     this._databaseConnector = new DatabaseConnector()
+
+    this.caches = {
+      'events': new Cache<Array<EventModel>>(this._databaseConnector,
```

Comment:
Why don't you use:
```suggestion
      events: new Cache<Array<EventModel>>(this._databaseConnector,
```
? So why do you need the single-quotes?